frontend/src/pages/projects/screens/spawner/useNotebookSizeState.ts

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import useNotebookDeploymentSize from '~/pages/projects/screens/detail/notebooks/useNotebookDeploymentSize';
+import { NotebookKind } from '~/k8sTypes';
+import { NotebookSize } from '~/types';
+import { getCustomNotebookSize } from '~/pages/projects/utils';
+import { useNotebookSize } from './useNotebookSize';
+
+export const useNotebookSizeState = (
+  existingNotebook: NotebookKind | undefined,
+): {
+  selectedSize: NotebookSize;
+  setSelectedSize: (size: string) => void;
+  sizes: NotebookSize[];
+} => {
+  const { size: notebookSize } = useNotebookDeploymentSize(existingNotebook);
+  const { selectedSize, setSelectedSize, sizes } = useNotebookSize();
+
+  const CUSTOM_NOTEBOOK_SIZE = getCustomNotebookSize(existingNotebook);
```

Comment:
This is going to trigger the `useEffect` on every render.

Reply:
resolving this as I'm not using `useEffect` anymore.

Reply:
It is still used as a dependency of `useMemo` in this hook. As an unstable reference, it forces all dependent hooks to be re-evaluated.

You'll want to utilize `useDeepCompareMemoize` on this.

Reply:
I'm computing when it's needed.